src/components/MovieListPage/MovieListPage.jsx

```diff
@@ -1,123 +1,56 @@
-  const handleMovieClick = (id) =>
-    setSelectedMovie(orderedMovies.find((movie) => movie.id === id));
-
-  useEffect(() => {
-    const controller = new AbortController();
-    const signal = controller.signal;
-
-    getMovies(
-      sortOption,
-      searchQuery,
-      selectedGenre === "All" ? "" : selectedGenre,
-      signal
-    )
-      .then((response) => setOrderedMovies(response.data))
-      .catch((error) => {
-        if (error.name !== "AbortError") {
-          console.error("Fetch error:", error.message);
-        }
-      });
-
-    return () => controller.abort();
-  }, [searchQuery, selectedGenre, sortOption]);
+import { useLocation, useNavigate } from "react-router-dom";
+import genres from "../../helpers/genres.js";
+import PropTypes from "prop-types";
+import React from "react";
+
+const MovieListPage = ({
+  genre,
+  sortBy,
+  handleGenre,
+  handleSortChange,
+  orderedMovies,
+}) => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const handleMovieClick = (movieId) =>
+    navigate(`/${movieId}${location.search}`);
 
   return (
     <>
-      <header className="App-header">
-        {!selectedMovie ? (
-          <div className="search-container">
-            <div className="blur-effect"></div>
-            <div className="content">
-              <div>
-                <button className="add-movie-button" onClick={toggleDialog}>
-                  {AddMovieButtonText}
-                </button>
-                <AddAndEditMovieDialog
-                  isOpen={isAddMovieDialogOpen}
-                  onClose={toggleDialog}
-                />
-              </div>
-              <Search
-                placeholder={SearchForm_Placeholder}
-                initialQuery={searchQuery}
-                onSearch={handleSearch}
-              ></Search>
-            </div>
-          </div>
-        ) : (
-          <div className="movie-details-container">
-            <MovieDetails
-              movie={selectedMovie}
-              onSearchClick={() => setSelectedMovie()}
-            />
-          </div>
-        )}
-      </header>
-
-      <div className="App-body">
-        <GenreSelector
-          genres={genres}
-          selectedGenre={genres[0]}
-          onSelect={handleGenre}
-        ></GenreSelector>
-        <SortControl
-          currentSelection={sortOption}
-          onSortChange={handleSortChange}
-        />
-
-        <div className="total-count">
-          <strong>{orderedMovies.length}</strong> movies found
-        </div>
-
-        <div className="App-movie-tiles">
-          {orderedMovies.map((movie, index) => (
-            <MovieTile
-              movie={movie}
-              key={index}
-              onClick={() => handleMovieClick(movie.id)}
-            />
-          ))}
-        </div>
+      <GenreSelector
+        genres={genres}
+        selectedGenre={genre}
+        onSelect={handleGenre}
+      ></GenreSelector>
```

Comment:
```suggestion
      />
```

You can use selfclose for components, that doesn't use `children` stuff

Reply:
updated